### PR TITLE
Atna udp interface

### LINKED ISSUE #375: Create the SYSLOG ATNA interface

### PR DESCRIPTION
**Pull request by devcritter**
Closes #375 

@rcrichton if you could please review? thanks

### CODE REVIEW

File: src/auditing.coffee [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    mergeAttrs: true,
+    explicitArray: false
+    tagNameProcessors: [firstCharLowerCase]
+    attrNameProcessors: [firstCharLowerCase, csdCodeToCode, originalTextToDisplayName]
+
+  parseString xml, options, (err, result) ->
+    return callback err if err
+
+    if not result?.auditMessage
+      return callback 'Document is not a valid AuditMessage'
```
Comment (rcrichton) [inline src/auditing.coffee]: Should we not create a new Error object here and set the message?

File: src/auditing.coffee [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  parseAuditRecordFromXML parsedMsg.message, (err, result) ->
+    audit = new Audit result
+
+    audit.rawMessage = msg
+    audit.syslog = parsedMsg
+    delete audit.syslog.originalMessage
+    delete audit.syslog.message
+
+    audit.save (saveErr) ->
+      if err or saveErr
```
Comment (rcrichton) [inline src/auditing.coffee]: Do we want to try save the audit if `err` exists?

**Comment by rcrichton** [comment]
@devcritter looks great! Just some minor comments above, otherwise :+1:

File: src/auditing.coffee (hunk 2 above)
Comment (devcritter) [inline src/auditing.coffee]: The use case I have in mind is, esp. with the UDP interface, if messages come through incomplete. That way we still store the raw message, even if we can't parse it. I saw this with a few times with OpenATNA and those big PnR messages. If we do get to the `err` stage, it also means it's valid syslog, so we won't be auditing garbage.

What do you think?

FYI looking at this code again, i see it's not too readable, just made a couple of small edits...

**Comment by devcritter** [comment]
@rcrichton made some changes

**Comment by rcrichton** [comment]
:+1:

File: src/auditing.coffee (hunk 2 above)
Comment (rcrichton) [inline src/auditing.coffee]: I see. I didn't get that at first, but I think it's a great idea.

**Pull request merged**